jaeger-core/src/test/java/com/uber/jaeger/ConfigurationTest.java

```diff
@@ -320,6 +321,53 @@ public void testPropagationValidFormat() {
     assertNotNull(textMap.get("uber-trace-id"));
   }
 
+  @Test
+  public void testSimplestBuilder() {
+    // there's no "easy" way to check what's the service name on a tracer, so, we'll just get one
+    // and assume it will blow up if a service name isn't provided
+    // to ensure it blows up, we have a test *without* a service name
+    Configuration.builder("Test").build().getTracer();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testNoServiceNameFails() {
+    Configuration.builder(null).build().getTracer();
+  }
+
+  @Test
+  public void testWithStatsReporter() {
+    InMemoryStatsReporter statsReporter = new InMemoryStatsReporter();
+
+    io.opentracing.Tracer tracer = Configuration
+          .builder("withStatsReporter")
+          .withStatsReporter(statsReporter)
+          .build()
+          .getTracer();
+    tracer.buildSpan("mySpan").start().finish();
+
+    // for now, we just get the first counter -- any available counter should be at 1 for this test
+    long counter = statsReporter.counters.values().iterator().next();
+    assertEquals(1, counter);
+  }
+
+  /**
+   * @see #testPropagationValidFormat() 
+   */
+  @Test
+  public void testPropagationViaBuilder() {
+    TestTextMap textMap = new TestTextMap();
+    SpanContext spanContext = new SpanContext(1234, 5678, 0, (byte)0);
+
+    Configuration
+          .builder("Test")
+          .withPropagationFormat("jaeger,invalid")
```

Comment:
I found builders with the prefix `with` (`TracerBuilder`, for instance) and without (`SenderConfiguration.Builder`). Which one should I use ?